### PR TITLE
Add TSDoc documentation

### LINKED ISSUE #5: Write extensive documentation
Write extensive documentation about the philosophy and usage of certain features and why they were standardized that way. Give developers and easy ride. Use doc-comments.

### PR DESCRIPTION
**Pull request by micahswitzer**
Resolves #5

### CODE REVIEW

**Comment by micahswitzer** [comment]
The rendered, HTML version of these docs are [here](https://micahswitzer.github.io/vmmx-schema/).

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: `[[Channels]]` ⇾ `[[Channel]]s`

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: `byt` ⇾ `by`

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: `prog.state.mute.snare = !prog.state.mute.snare` ⇾ `prog.state.machine.mute.snare = !prog.state.machine.mute.snare`

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: Perhaps clarify `All the events that occur...` to `All the [[TickedDropEvent]]s that occur...`

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: This implies that performance is inside a program. Perhaps change it to `Represents the performance on a [[Program]].`

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: Confusing to read. Perhaps change it to `Alternatively, a bunch of machine events could be specified for tick 0, but this way is simpler.`
May even be expunged as it's pretty trivial.

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: `[[Program.events]]` ⇾ `[[Program.dropEvents]]`

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: Perhaps clarify `This equates the the programming wheel starting or stopping.` ⇾ `This equates to the programming wheel starting or continuing instantly.`
(also note the `the` ⇾ `to`)

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: This is supposed to be a multiplier of some sort.
Currently, no fancy gear-to-speed conversion is planned. This may change when it's implemented, however.

For now, it should just be a 0-1 multiplier of vibrato speed.

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: `throughout the course of an entire program.` ⇾ `throughout the course of an entire performance.`
As you can change when programming, but not when performing.

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: Which `E`, `A`, `D` and `G`?
Definitely no musician but isn't there a number missing that indicates the "highness" on a note?

File: src/schema.ts
Comment (mozi-h) [inline src/schema.ts]: Perhaps no shortcuts as to avoid confusion.

**Review by mozi-h — CHANGES_REQUESTED** [review]
Great work!
I've only found minor clarifications and typos.

File: src/schema.ts
Comment (micahswitzer) [reply src/schema.ts]: Okay. I think if that's the case, we shouldn't have a separate `vibratoEnabled` property when you could simply set `vibratoSpeed` to 0. But this is conversation we can carry on somewhere else.

File: src/schema.ts
Comment (iansedano) [inline src/schema.ts:97]: Maybe some comments for each state
```
machine: MachineState; // channel mutes, bpm, flywheel connected
vibraphone: VibraphoneState; // vibrato enabled, vibrato speed (also dampening when implemented)
bass: BassState; // Capo positions and tuning
hihatMachine: HihatMachineState; // setting (whole note / offbeat / 16ths / triplet) (not specified in schema yet)
hihat: HihatState; // closed or open
```

File: src/schema.ts
Comment (iansedano) [inline src/schema.ts:152]: Maybe a lay explanation
```
* A Performance is considered as events that are outside of the main programming wheel.
* i.e. events that can affect what the programming wheel plays across
* various revolutions of the programming wheel.
* For example, you have the drums and bass muted
* for the whole first revolution of the programming wheel.
```

File: src/schema.ts
Comment (iansedano) [inline src/schema.ts:205]: Some extra comments maybe?
		| DropEvent // Performances only store manual drops
		| MachineEvent // mutes
		| VibraphoneEvent // vibrato (& dampener when implemented)
		| HihatMachineEvent // setting
		| HihatEvent // open or closed
		| BassEvent; // capo change

Although this may make things harder to update if things change...

File: src/schema.ts
Comment (iansedano) [inline src/schema.ts:213]: I got really confused here for a while thinking that it was referring to the bass, so maybe a quick note that says "Not Bass instrument - Base absolute time" or something to that effect?

File: src/schema.ts
Comment (iansedano) [inline src/schema.ts:231]: Why does MachineEvent = MachineMuteEvent but not also MachineTempoEvent or FlywheelConnectedEvent?

File: src/schema.ts
Comment (tilleraj) [inline src/schema.ts:315]: Should clarify if 1 is the low string or the high string (closest to the machine or furthest away). It's assumed/conventional that 1 is low but it's not explicit

File: src/schema.ts
Comment (tilleraj) [inline src/schema.ts:383]: May make more sense if changed to `HihatStateEvent` as the open/close state of the hihat is called `HihatState` and not `HihatClosed`

File: src/schema.ts
Comment (mozi-h) [reply src/schema.ts]: Me too; there was some back and forth on this and we settled here. This can be dealt with by whomever implements vibrato 🤷‍♂️.

File: src/schema.ts
Comment (mozi-h) [reply src/schema.ts:97]: If people want to know what properties the different states have, they can simply look it up in the specific State definition.

File: src/schema.ts
Comment (mozi-h) [reply src/schema.ts:205]: If people want to know what properties the different events have, they can simply look it up in the specific Event definition.
~`Performances only store manual drops` might be good to add, but in the Program's doc string.~

File: src/schema.ts
Comment (mozi-h) [reply src/schema.ts:152]: ~\+ `Performances only store manual drops`~
This was thought over; see PR #14.

File: src/schema.ts
Comment (mozi-h) [reply src/schema.ts:231]: Nice catch! This is an oversight. We updated the `kind`s, but forgot about the name of `FlywheelConnectedEvent` and adding it here 👏

File: src/schema.ts
Comment (mozi-h) [reply src/schema.ts:383]: With the naming norm we try to tell a bit more about what that event changes. It's always called `...State`, so this might be a very generic name.

**Review by mozi-h — APPROVED** [review]
Will wait until the other review comments are resolved (and re-reviewed) before merging.

File: src/schema.ts
Comment (ollpu) [reply src/schema.ts:213]: Hmm, maybe we should just change the name. There's `CoreDropEvent` and this. There's no semantic reason to name them differently, right?

File: src/schema.ts
Comment (mozi-h) [reply src/schema.ts:205]: Update: There is actually an issue with this approach; see PR #14.

File: src/schema.ts
Comment (micahswitzer) [reply src/schema.ts:213]: This is addressed in #14.

**Comment by micahswitzer** [comment]
@mozi-h I've implemented all the documentation-specific changes and am ready for merging to happen.

**Pull request merged**
